packages/back-end/src/routers/environment/environment.controller.ts

```diff
@@ -18,18 +26,88 @@ type CreateEnvironmentResponse = {
   environment: Environment;
 };
 
+export const putEnvironmentOrder = async (
+  req: AuthRequest<{
+    environments: string[];
+  }>,
+  res: Response
+) => {
+  const context = getContextFromReq(req);
+  const { org } = context;
+  const envIds = req.body.environments;
+
+  const existingEnvs = org.settings?.environments;
+
+  if (!existingEnvs) {
+    return res.status(400).json({
+      status: 400,
+      message: "Unable to find organization's environments",
+    });
+  }
+
+  // If the user doesn't have permission to update any envs, don't allow this action
+  if (
+    existingEnvs.every(
+      (env) => !context.permissions.canCreateOrUpdateEnvironment(env)
+    )
+  ) {
+    context.permissions.throwPermissionError();
+  }
+
+  const updatedEnvs: Environment[] = [];
+
+  // Loop through env ids, to get the full env object and add it to the updatedEnvs arr
+  envIds.forEach((environment) => {
+    const index = existingEnvs.findIndex((env) => env.id === environment);
+
+    if (index < 0) {
+      return res.status(400).json({
+        status: 400,
+        message: `Unable to find environment: ${environment}`,
+      });
+    }
+
+    updatedEnvs.push(existingEnvs[index]);
```

Comment:
Any reason for now using the plain `find` method?

Reply:
Looks like the same code as `putEnvironments`. Maybe it could be factored out?

Reply:
No good reason for using `find` vs `findIndex`. I'll update the `find` as it does make things a bit more concise.

Reply:
Actually, I think for the `putEnvironment` it makes sense to keep the `findIndex` as with that, we want to find the index, so we can later update that specific index. I think `find` is better for the `putEnvironmentOrder` and `deleteEnvironment`, though.